### DISCUSSION TITLE
[Enhancement] [input] Add a word-length property to the input component to support a custom method for calculating the length of characters

### DISCUSSION THREAD

**Discussion by grzhan** [element-plus/element-plus#18940] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `input`

## Description

Currently, the el-input component supports the maxlength and minlength properties to limit the number of input characters. However, it does not support customizing how the character length is calculated. For Chinese characters, the current implementation treats each Chinese character as one character, but in some scenarios, we might want to calculate the length based on byte count or in a more customized way for non-ASCII characters.

One possible implementation is to add a word-length property that accepts a function of type (value: string) => number, allowing for custom character length calculation methods.

### Living Examples

The Input component in Arco Design Vue supports the word-length property, which allows for customizing the method of calculating character length:
https://arco.design/vue/en-US/component/input#API

### Additional comments

_(empty)_

**Comment by li936393** [comment]
Our project has similar needs and hope to support it as soon as possible

**Comment by EZ105** [comment]
The feature is really needed.

**Comment by Liao-js** [comment]
The el-input component supports maxlength and minlength attributes which are native attributes of input.I know how to implement the native maxlength, but I don't know how to implement the native minlength.
